AzureWebFarm.OctopusDeploy/Infrastructure/OctopusDeploy.cs

```diff
@@ -45,8 +47,51 @@ public void ConfigureTentacle()
             _processRunner.Run(_tentaclePath, string.Format("service {0} --install --start --console", InstanceArg));
         }
 
+        private void HandleFailedRemoval()
+        {
+            if (File.Exists(Path.Combine(_tentacleInstallPath, "Tentacle.config")))
+            {
+                try
+                {
+                    Log.Information("Detected existing tentacle.config - possible abrupt restart. Deleting tentacle.config.");
+                    File.Delete(Path.Combine(_tentacleInstallPath, "Tentacle.config"));
+                    Log.Information("Successfully removed tentacle.config from role.");
+                }
+                catch (Exception e)
+                {
+                    Log.Error(e, "Failed to remove certificate from Octopus server. Manual intevention required.");
+                }
+
+                try
+                {
+                    Log.Information("Attempting to remove tentacle from Octopus server.");
+                    DeleteMachine();
+                    Log.Information("Successfully removed tentacle from Octopus server.");
+                }
+                catch (Exception e)
+                {
+                    Log.Error(e, "Failed to remove tentacle from Octopus server. Manual intevention required.");
+                }
+
+            }
+        }
+
         public void DeleteMachine()
         {
+            try
+            {
+                DeleteMachineInternal();

```

Comment:
Another option would be to use something like Polly to retry with a strategy (e.g. exponential backoff), but happy with this if it proves to be robust - let's try it for now :)